### PR TITLE
GRW-713/New-Inputs

### PR DESCRIPTION
**Pull request by dendoesit**
## What?

This will be new branch for input fields + errors .  We consider the old one with all the comments for inputs as approved. https://github.com/HedvigInsurance/web-onboarding/pull/732

Update to input fields.
Add generic errors to detailsModal and checkout 

## Why?

The new styling for input fields needs error messages instead of using the defaults from Formik. 

## **NIT**
 Worth noting that we discussed we will not focus on fixing all errors and issues on the current OfferNew or DetailsModal as we plan on moving to the new Offer soon. 
**Ticket(s): [https://hedvig.atlassian.net/browse/GRW-713]**

## Testing
Most changes can be noticed in the Checkout and inside the Details Modal both on the old and new debugger but every input on the website should be affected.

Before:
image

After: 
image

### [Review app]

https://web-onboardi-grw-713-fe-bs8s79.herokuapp.com/se-en/new-member/debugger

### CODE REVIEW

File: src/client/pages/OfferNew/Introduction/DetailsModal/utils.ts [hunk 1]
```diff
@@ -221,7 +230,7 @@ const getSwedishSchema = (
                 label: getExtraBuilding(value),
                 value,
               })),
-              validation: Yup.string().required(),
+              validation: Yup.string(),
```
[the remark this answers is not in the document]
Comment (dendoesit) [reply src/client/pages/OfferNew/Introduction/DetailsModal/utils.ts:233]: Required fields are not needed on select dropdowns as we currently always take the first option in a select.

**Review by mehdielket — DISMISSED** [review]
Looks good to me. I tested a bit in the review app and I think it is ok as it is. 

I think we should be very careful when deploying this and test it thoroughly

**Review by robinandeer — COMMENTED** [review]
Hmm there's a couple of empty files that we should maybe remove:

- src/client/components/inputs/types.ts
- src/client/components/inputs/utils.ts

File: src/client/pages/Forever/components/RedeemCode.tsx [hunk 2]
```diff
@@ -185,7 +185,6 @@ export const RedeemCode: React.FC<RedeemCodeProps> = ({
                   }
                   name="code"
                   type="text"
-                  autoComplete="off"
```
Comment (robinandeer) [inline src/client/pages/Forever/components/RedeemCode.tsx:188]: Why do we change this?

File: src/client/pages/LoginApp/components/LoginForm.tsx [hunk 3]
```diff
@@ -122,7 +122,6 @@ export const LoginForm: React.FC<LoginFormProps> = ({ onSubmit }) => {
                 id="ssn"
                 type="text"
                 inputMode="numeric"
-                pattern="[0-9]*"
```
Comment (robinandeer) [inline src/client/pages/LoginApp/components/LoginForm.tsx:125]: Why change this?

File: src/client/pages/Offer/Introduction/DetailsModal/Details/components/DetailInput.tsx [hunk 4]
```diff
@@ -47,7 +46,7 @@ export const DetailInput: React.FC<DetailInputProps &
       showErrorMessage={false}
       errors={getIn(formikProps.errors, formikName)}
       touched={getIn(formikProps.touched, formikName)}
-      autoComplete="off"
+      {...formikProps.getFieldProps(formikName)}
```
Comment (robinandeer) [inline src/client/pages/Offer/Introduction/DetailsModal/Details/components/DetailInput.tsx:49]: Does this mean we pass "autoComplete" to formik instead?

File: src/client/pages/LoginApp/components/LoginForm.tsx (hunk 3 above)
Comment (dendoesit) [reply src/client/pages/LoginApp/components/LoginForm.tsx:125]: We already have inputMode numeric on this which only allows characters between 0 and 9 so this is not needed.

File: src/client/pages/Forever/components/RedeemCode.tsx (hunk 2 above)
Comment (dendoesit) [reply src/client/pages/Forever/components/RedeemCode.tsx:188]: It is hard for me to see a scenario where your browser already knows the redeemed code for this so it actually uses the autoComplete option.

File: src/client/pages/Offer/Introduction/DetailsModal/Details/components/DetailInput.tsx (hunk 4 above)
Comment (dendoesit) [reply src/client/pages/Offer/Introduction/DetailsModal/Details/components/DetailInput.tsx:49]: I have removed autoComplete to simplify code as I couldn't see it bring any value. 
Furthermore Chrome ignores the autoComplete='off' from version 34. https://stackoverflow.com/questions/31612867/autocomplete-off-not-working-in-chrome

I am not sure if Formik handle autoComplete.

**Comment by dendoesit** [comment]
@robinandeer to me it looks like the files have been removed. 
image

File: src/client/pages/Forever/components/RedeemCode.tsx (hunk 2 above)
Comment (robinandeer) [reply src/client/pages/Forever/components/RedeemCode.tsx:188]: Hmm.. I think the browser might decide to suggest any number of irrelevant things so I would keep "autoComplete" turned off.

File: src/client/pages/LoginApp/components/LoginForm.tsx (hunk 3 above)
Comment (robinandeer) [reply src/client/pages/LoginApp/components/LoginForm.tsx:125]: Seems like Safari (desktop) doesn't support "inputMode" => so perhaps we need to keep it for that reason. But could be good to add that as a comment.

https://caniuse.com/input-inputmode

File: src/client/pages/Offer/Introduction/DetailsModal/Details/components/DetailInput.tsx (hunk 4 above)
Comment (robinandeer) [reply src/client/pages/Offer/Introduction/DetailsModal/Details/components/DetailInput.tsx:49]: Hmm interesting. I guess it's not a standard. I'm okey not using it.

**Comment by robinandeer** [comment]
> @robinandeer to me it looks like the files have been removed. image

Oh sorry about that. Must've read it wrong.

File: src/client/pages/LoginApp/components/LoginForm.tsx (hunk 3 above)
Comment (dendoesit) [reply src/client/pages/LoginApp/components/LoginForm.tsx:125]: oh Safari 😦 . great catch 👍 .

**Review by robinandeer — APPROVED** [review]
Amazing patience and work @dendoesit!

Things behave as expected now in the review app as far as I can tell. Sorry for delaying the code review - but with these huge PRs I just get overwhelmed 😪

Still feel like the spacing is a bit off (especially in the edit details modal) -- but we can discuss that will Amy/Zak later

**Pull request merged**